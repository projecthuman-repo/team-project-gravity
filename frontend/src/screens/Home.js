import React, { useState } from "react";
import {View, Text, TouchableWithoutFeedback, Image, Alert, Platform, SafeAreaView} from "react-native";
import Auth0 from 'react-native-auth0';
import useUser from '../hooks/queries/useUser';

import * as RA from 'ramda-adjunct';

import Styles from "../style/Style";

export default function Home({ navigation }) {
    
    // SUPRESS WARNINGS
    console.disableYellowBox = true;

    const [accessToken, setAccessToken] = useState('');
    const [userID, setUserID] = useState('');
    const mobile_credentials = require('../auth0-configuration-mobile');
    const auth0 = new Auth0(mobile_credentials);

    const checkUserExists = () => {
        if (user.bio) {
            navigation.navigate("CommunityList", {userID: userID})
        } else {
            navigation.navigate("Picture", {userID: userID})
        }
    }

    const user = useUser(userID);
    if (RA.isNotNil(user)) {
        checkUserExists();
    }

    const goToCommunityList = () => {
        //TODO: update userId to be something like "guest" which indicates no one has signed in
        // then anywhere that Platform.OS is used, replace with check to see if userId === 'guest'
        navigation.navigate("CommunityList")
    }


    //micas for testing so remove after
    const goToPicture = () => {
        navigation.navigate("Picture",  {userID: "12"})
    }

    //micas for testing so remove after
    const dummyuid = () => {
        navigation.navigate("CommunityList",  {userID: "1"})
    }

    const getUserID = async () => {
        const url = `https://${mobile_credentials.domain}/userinfo`;

        var header = new Headers();
        header.append('Authorization', `Bearer ${accessToken}`);
        header.append('Accept', 'application/json');

        const request = new Request(url, {
            method: 'GET',
            headers: header,
        });
        const res = await fetch(request);
        try {
            console.log(res.status);
            const userInfo = await res.json();
            return userInfo.sub;
        } catch(err) {
            console.log("JSON ERR");
            console.log(err);
        }
    }

    const onLogin = async () => {
        try {
            // await auth0.webAuth.clearSession({});
            const credentials = await auth0.webAuth.authorize({scope: 'openid profile email'});
            setAccessToken(credentials.accessToken);

            const subUserID = await getUserID();

            setUserID(subUserID.replace("auth0|", ""));
            checkUserExists();

        } catch(err) {
            console.log(err);
        }
    };
    
    // The text things are for spaces, not sure of a better way to do it
    return(
        <SafeAreaView style={Styles.SafeAreaViewStyle}>
            <View style={Styles.MiddleOfScreen}>
                <View style={Styles.logoContainer}>
                    <Image style={Styles.logo} source={require('../images/logo.jpeg')}></Image>
                </View>

                {Platform.OS === 'web' ? null :
                <TouchableWithoutFeedback onPress={() => onLogin()}>
                    <View style={Styles.Button}>
                        <Text style={Styles.ButtonText}> LOGIN </Text>
                    </View>
                </TouchableWithoutFeedback>
                }
                <TouchableWithoutFeedback onPress={goToCommunityList}>
                    <View style={Styles.Button}>
                        <Text style={Styles.ButtonText}> CONTINUE AS GUEST </Text>
                    </View>
                </TouchableWithoutFeedback>
                {//micas for testing so remove after 
                }               
                {/* <TouchableWithoutFeedback onPress={dummyuid}>
                    <View style={Styles.Button}>
                        <Text style={Styles.ButtonText}> FOR TESTING </Text>
                    </View>
                </TouchableWithoutFeedback> */}
            </View>
        </SafeAreaView>
    )
}
