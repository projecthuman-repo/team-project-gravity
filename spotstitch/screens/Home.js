import React from "react";
import {View, Text, TouchableWithoutFeedback, Image} from "react-native";

import Styles from "../style/Style";

export default function Home({ navigation }) {
    
    const goToLogin = () => {
        navigation.navigate("Login")
    }

    const goToSignup = () => {
        navigation.navigate("Signup")
    }

    const goToCommunity = () => {
        navigation.navigate("Community")
    }
    
    // The text things are for spaces, not sure of a better way to do it
    return(
        <View style={Styles.MiddleOfScreen}>
            
            <Text style={Styles.Title}> Home Page </Text>
            <View style={Styles.logoContainer}>
                <Image style={Styles.logo} source={require('../assets/logo.jpeg')}></Image>
            </View>

            <TouchableWithoutFeedback onPress={goToLogin}>
                <View style={Styles.Button}>
                    <Text style={Styles.ButtonText}> LOGIN </Text>
                </View>
            </TouchableWithoutFeedback>

            <TouchableWithoutFeedback onPress={goToSignup}>
                <View style={Styles.Button}>
                    <Text style={Styles.ButtonText}> SIGN UP </Text>
                </View>
            </TouchableWithoutFeedback>
            
            <TouchableWithoutFeedback onPress={goToCommunity}>
                <View style={Styles.Button}>
                    <Text style={Styles.ButtonText}> SEE A COMMUNITY </Text>
                </View>
            </TouchableWithoutFeedback>
        </View>
    )
}