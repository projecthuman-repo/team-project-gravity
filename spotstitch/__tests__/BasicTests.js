import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../screens/Home.js';

test('renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });