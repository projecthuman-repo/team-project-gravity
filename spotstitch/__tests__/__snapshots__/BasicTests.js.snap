// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/logo.jpeg",
        }
      }
      style={
        Object {
          "height": 267,
          "width": 300,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fa5f6a",
        "borderRadius": 5,
        "fontFamily": "Asap-Regular",
        "marginBottom": 10,
        "padding": 10,
        "width": 375,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Asap-Regular",
        }
      }
    >
       LOGIN 
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fa5f6a",
        "borderRadius": 5,
        "fontFamily": "Asap-Regular",
        "marginBottom": 10,
        "padding": 10,
        "width": 375,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Asap-Regular",
        }
      }
    >
       SIGN UP 
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fa5f6a",
        "borderRadius": 5,
        "fontFamily": "Asap-Regular",
        "marginBottom": 10,
        "padding": 10,
        "width": 375,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Asap-Regular",
        }
      }
    >
       SEE A COMMUNITY 
    </Text>
  </View>
</View>
`;
